modular_skyrat/modules/gunsgalore/code/guns/master.dm

```diff
@@ -0,0 +1,244 @@
+
+/obj/item/gun/ballistic/can_shoot()
+	if(realistic)
+		if(jammed)
+			return FALSE
+		else
+			. = ..()
+	else
+		. = ..()
+
+/obj/item/gun/ballistic/process_burst(mob/living/user, atom/target, message, params, zone_override, sprd, randomized_gun_spread, randomized_bonus_spread, rand_spr, iteration)
+	if(realistic)
+		if(jammed)
+			firing_burst = FALSE
+			return FALSE
+		else
+			. = ..()
+	else
+		. = ..()
+
+/obj/item/gun/ballistic/process_fire()
+	if(realistic)
+		if(jammed)
+			shoot_with_empty_chamber()
+			return
+	. = ..()
+
+/obj/item/gun/ballistic/shoot_live_shot(mob/living/user, pointblank, atom/pbtarget, message)
+	if(realistic)
+		if(jammed)
+			return FALSE
+
+		dirt_level += dirt_modifier
+
+		durability = clamp(durability -= durability_factor, 1, 1000)
+
+		jam_chance = dirt_level/5
+
+		spread = base_spread + ((jam_chance / durability)*100)
+
+		switch(FLOOR(durability, 1))
+			if(0 to 9)
+				if(prob(90))
+					jam(user)
+					return FALSE
+			if(10 to 29)
+				if(prob(10))
+					jam(user)
+					return FALSE
+			if(30 to 49)
+				if(prob(5))
+					jam(user)
+					return FALSE
+
+		if(dirt_level > 30 && prob(jam_chance))
+			jam(user)
+			return FALSE
+	. = ..()
+
+/obj/item/gun/ballistic/AltClick(mob/user)
+	if(realistic)
+		if(!user.canUseTopic(src))
+			return
+		if(jammed)
+			jam(TRUE, user)
+			return
+	. = ..()
+
+/obj/item/gun/ballistic/afterattack(atom/target as mob|obj|turf, mob/living/user as mob|obj, flag, params)
+	if(realistic)
+		if(jammed)
+			to_chat(user, "<span class='warning'>[src] is jammed, unjam it before firing!</span>")
+			return
+		else
+			. = ..()
+			update_icon()
+	else
+		. = ..()
+
+/obj/item/gun/ballistic/examine(mob/user)
+	. = ..()
+	if(realistic)
+		switch(FLOOR(dirt_level, 1))
+			if(0 to 10)
+				. += "It looks clean."
+			if(11 to 30)
+				. += "It looks slightly dirty."
+			if(31 to 50)
+				. += "It looks dirty."
+			if(51 to 70)
+				. += "It looks very dirty."
+			else
+				. += "<span class='warning'>It is filthy!</span>"
+
+		switch(FLOOR(durability, 1))
+			if(0 to 9)
+				. += "<span class='warning'><b>It is falling apart!</b></span>"
+			if(10 to 29)
+				. += "<span class='warning'>It looks battle scarred!</span>"
+			if(30 to 49)
+				. += "It looks well worn."
+			if(50 to 69)
+				. += "It has minimal wear."
+			else
+				. += "It looks factory new."
+
+		if(jammed)
+			. += "<span class='warning'><b>It is jammed, alt+click it to unjam it!</b></span>"
+		else if(durability < 10)
+			. += "<span class='warning'><b>It is barely functioning!</b></span>"
+		else
+			. += "It is functioning normally."
+
+/obj/item/gun/ballistic/attackby(obj/item/A, mob/user, params)
+	. = ..()
+	if(realistic)
+		if(istype(A, /obj/item/soap))
+			var/obj/item/soap/S = A
+			to_chat(user, "<span class='notice'>You start cleaning the [src].</span>")
+			if(do_after(user, 30 SECONDS))
+				dirt_level -= S.cleanspeed
+				if(dirt_level < 0)
+					dirt_level = 0
```

Comment:
Perhaps clean speed should affect the delay time? And then it'd be cleaned to 0

Reply:
Hmm, well, 30 seconds is a nice amount of time to delay someone and make it so that cleaning weapons is something you do after battle